frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/new-experiment/new-experiment.component.ts

```diff
@@ -107,11 +104,13 @@ export class NewExperimentComponent implements OnInit {
 
   checkPostExperimentRule() {
     if (this.newExperimentData.postExperimentRule === POST_EXPERIMENT_RULE.ASSIGN) {
-      const conditionToSet = this.newExperimentData.conditions.filter((condition) => {
-        condition.id === this.newExperimentData.revertTo;
-      });
-      if (!conditionToSet.length) {
-        this.newExperimentData.postExperimentRule = POST_EXPERIMENT_RULE.CONTINUE;
+      if (this.newExperimentData.revertTo) {
```

Comment:
This `if` condition can directly be checked in the above `if` condition:
`if (this.newExperimentData.postExperimentRule === POST_EXPERIMENT_RULE.ASSIGN && this.newExperimentData.revertTo) {`

Reply:
fixed